__tests__/event_manager.test.ts

```diff
@@ -16,20 +16,72 @@ describe('Event Handler', function() {
     ain.em.disconnect();
   });
 
-  it('Subscribe to BLOCK_FINALIZED', (done) => {
-    ain.em.subscribe('BLOCK_FINALIZED', {
-      block_number: null,
-    }, (data) => {
-      done();
+  describe('BLOCK_FINALIZED', () => {
+    it('Subscribe to BLOCK_FINALIZED', (done) => {
+      ain.em.subscribe('BLOCK_FINALIZED', {
+        block_number: null,
+      }, (data) => {
+        done();
+      });
+    });
+
+    it('Subscribe to BLOCK_FINALIZED with wrong config', (done) => {
+      ain.em.subscribe('BLOCK_FINALIZED', {
+        block_number: -1,
+      }, (data) => {
+      }, (err) => {
+        done();
+      });
     });
   });
 
-  it('Subscribe to BLOCK_FINALIZED with wrong config', (done) => {
-    ain.em.subscribe('BLOCK_FINALIZED', {
-      block_number: -1,
-    }, (data) => {
-    }, (err) => {
-      done();
+  describe('VALUE_CHANGED', () => {
+    const [ testAccount ] = ain.wallet.create(1);
+    const testAppName = `test_${Date.now()}`;
+    const testAppPath = `/apps/${testAppName}`;
+
+    beforeAll(async () => {
+      // Create test app
+      ain.wallet.setDefaultAccount(testAccount);
+      await ain.db.ref(`/manage_app/${testAppName}/create/${Date.now()}`).setValue({
+        value: { admin: { [testAccount]: true } },
+      });
+    });
+
+    it('Subscribe to VALUE_CHANGED', (done) => {
+      let blockEventCheck = false;
+      let userEventCheck = false;
+
+      ain.em.subscribe('VALUE_CHANGED', {
+        path: testAppPath,
+        event_source: null,
```

Comment:
could you add test cases with 'BLOCK' and 'USER'?

Reply:
Done!